pkg/blockbuilder/blockbuilder.go

```diff
@@ -416,16 +415,22 @@ func (b *BlockBuilder) shipperDir() string {
 	return filepath.Join(b.cfg.BlocksStorageConfig.TSDB.Dir, "shipper")
 }
 
-func (b *BlockBuilder) commitOffset(ctx context.Context, part int32, offset int64) (returnErr error) {
+func (b *BlockBuilder) commit(ctx context.Context, rec *kgo.Record, lastOffset, currEnd int64) (returnErr error) {
 	defer func() {
 		if returnErr != nil {
-			level.Error(b.logger).Log("msg", "failed to commit offset to Kafka", "err", returnErr, "partition", part, "offset", offset)
+			level.Error(b.logger).Log("msg", "failed to commit offset to Kafka", "err", returnErr, "partition", rec.Partition, "offset", rec.Offset)
 			// TODO(codesome): add metric
 		}
 	}()
 
 	toCommit := kadm.Offsets{}
-	toCommit.AddOffset(b.cfg.Kafka.Topic, part, offset, -1)
+	toCommit.Add(kadm.Offset{
+		Topic:       rec.Topic,
+		Partition:   rec.Partition,
+		At:          rec.Offset,
+		LeaderEpoch: -1,
+		Metadata:    marshallCommitMeta(lastOffset, currEnd),
+	})
```

Comment:
Have you tried this against real Kafka? In #8296 I replaced it with not going via `kadm.Offset` directly, because this didn't work, failing on commit.

I believe, `LeaderEpoch` cannot be -1 when the consumer is part of a consumer group. Refer to [Kafka protocol spec][1]:

> Note that when this API is used for a "simple consumer," which is not part of a consumer group, then the generationId must be set to -1 and the memberId must be empty (not null). Additionally, if there is an active consumer group with the same groupId, then the commit will be rejected (typically with an `UNKNOWN_MEMBER_ID` or `ILLEGAL_GENERATION` error). 

[1]: https://cwiki.apache.org/confluence/display/KAFKA/A+Guide+To+The+Kafka+Protocol#AGuideToTheKafkaProtocol-OffsetCommitRequest